### PR TITLE
fix(subgouplink): recursively get subgoup links

### PR DESCRIPTION
**Pull request by qingyashu**
### Bug Fixes
Recursively get subgroup links when building graph

### CODE REVIEW

File: src/GraphUtils/utils.js [hunk 1]
```diff
@@ -1,3 +1,29 @@
...
+  let subgroupLinks = [];
+  if (link.subgroup) {
+    link.subgroup.forEach((sgLink) => {
+      if (sgLink.subgroup) {
+        subgroupLinks = subgroupLinks.concat(getSubgroupLinks(sgLink, nameToNode, sourceId));
+      } else {
+        subgroupLinks.push({
+          source: nameToNode[sourceId],
+          target: nameToNode[sgLink.target_type],
+          exists: 1,
```
Comment (abgeorge7) [inline src/GraphUtils/utils.js]: what is "exists" for?

Comment (qingyashu) [reply src/GraphUtils/utils.js]: that's for previous logic in `createNodesAndEdges`, I think it's used for filtering out 0 count links for data model graph.

File: src/GraphUtils/utils.js [hunk 2]
```diff
@@ -1,3 +1,29 @@
+/**
+ * Get subgroup links from link
+ * @param {array} link - array of links
```
Comment (abgeorge7) [inline src/GraphUtils/utils.js]: this is picky but usually if something is an array it feels like it should be plural, so `links` instead of `link`

Comment (abgeorge7) [reply src/GraphUtils/utils.js]: The code seems like `link` is not an array but an object though?

**Comment by qingyashu** [comment]
yeah it's typo... fixed

**Pull request merged**